crypto/kdf/hkdf.c

```diff
@@ -245,7 +283,7 @@ static unsigned char *HKDF_Expand(const EVP_MD *evp_md,
     if (okm_len % dig_len)
         n++;
 
-    if (n > 255)
+    if (n > 255 || okm == NULL)
```

Comment:
Ooops yeah, I got confused by the okm_len variable just above.  So this is fine.